admin/src/screens/burgers/burger.state.ts

```diff
@@ -0,0 +1,238 @@
+      burgersList: listSorted,
+    });
+  }, []);
+
+  useEffect(() => {
+    fetchBurgerList();
+  }, [fetchBurgerList]);
+
+  const fetchIngredientsInBurgerList = useCallback(async () => {
+    if (state.activeBurger) {
+      const { data } = await trackPromise(
+        getIngredientsInBurger(state.activeBurger.id),
+        PROMISES_AREA.getIngredientsInBurgerList
+      );
+      updateState({
+        ingredientsInBurgerList: data,
+      });
+    }
+  }, [state.activeBurger]);
+
+  useEffect(() => {
+    if (state.isEditForm && state.isBurgerFormActive) {
+      fetchIngredientsInBurgerList();
+    }
+  }, [
+    fetchIngredientsList,
+    fetchIngredientsInBurgerList,
+    state.isEditForm,
+    state.isBurgerFormActive,
+  ]);
+
+  const openCreateForm = () => {
+    updateState({
+      isEditForm: false,
+      isBurgerFormActive: true,
+      activeBurger: null,
+    });
+  };
+
+  const closeForm = () => {
+    updateState({
+      isBurgerFormActive: false,
+      activeBurger: null,
+    });
+  };
+
+  const addBurger = async (
+    values: IBurgerValues,
+    { resetForm }: FormikHelpers<IBurgerValues>
+  ) => {
+    const { data } = await trackPromise(
+      createBurger({ ...values, price: +values.price }),
+      PROMISES_AREA.addBurger
+    );
+
+    const newList = [...state.burgersList];
+    newList.push(data);
+
+    updateState({
+      burgersList: newList,
+    });
+    resetForm();
+    closeForm();
+  };
+
+  const onItemClick = (id: string) => {
+    updateState({
+      activeBurger: state.burgersList.find((i) => i.id === id) || null,
+      isBurgerFormActive: true,
+      isEditForm: true,
+    });
+  };
+
+  const removeBurger = async (id: string) => {
+    const removed = await deleteBurger(id);
+    const filteredList = [...state.burgersList].filter((item) => item.id !== id);
+
+    updateState({
+      burgersList: filteredList,
+    });
+
+    closeForm();
+    return removed;
+  };
+
+  const editIngredient = async (id: string, body: ICreateBurger) => {
+    const { data } = await updateBurger(id, body);
+    const newList = [...state.burgersList].map((item) => {
+      if (item.id === data.id) {
+        return data;
+      }
+
+      return item;
+    });
+
+    updateState({
+      burgersList: newList,
+    });
+
+    return data;
+  };
+
+  const addIngredientInBurger = async (body: ICreateIngredientInBurger) => {
+    const { data } = await trackPromise(
+      createIngredientInBurger(body),
+      PROMISES_AREA.addIngredientInBurger
+    );
+
+    const newList = [...state.ingredientsInBurgerList];
+    newList.push(data);
+
+    updateState({
+      ingredientsInBurgerList: newList,
+      isShowModal: false,
+    });
+
+    return data;
+  };
+
+  const editIngredientInBurger = async (
+    id: string,
+    body: IUpdateIngredientInBurger
+  ) => {
+    const { data } = await updateIngredientInBurger(id, body);
+
+    const newList = [...state.ingredientsInBurgerList].map((item) => {
```

Comment:
why do not map `state.ingredientsInBurgerList`? There is no need to do suck destruction